arcgis-ios-sdk-samples/Utility Network/Display utility association/DisplayUtilityAssociationVC.swift

```diff
@@ -0,0 +1,141 @@
+                edges.filter { $0.sourceUsageType != .subnetLine }.forEach { source in
+                    self.mapView.map?.operationalLayers.add(AGSFeatureLayer(featureTable: source.featureTable))
+                }
+                // Add all the junctions to the map.
+                junctions.forEach { source in
+                    self.mapView.map?.operationalLayers.add(AGSFeatureLayer(featureTable: source.featureTable))
+                }
+                
+                // Create a renderer for the associations.
+                let attachmentValue = AGSUniqueValue(description: "Attachment", label: "", symbol: self.attachmentSymbol, values: [AGSUtilityAssociationType.attachment])
+                let connectivityValue = AGSUniqueValue(description: "Connectivity", label: "", symbol: self.connectivitySymbol, values: [AGSUtilityAssociationType.connectivity])
+                self.associationsOverlay.renderer = AGSUniqueValueRenderer(fieldNames: ["AssociationType"], uniqueValues: [attachmentValue, connectivityValue], defaultLabel: "", defaultSymbol: nil)
+                
+                // Populate the legened.
+                self.attachmentSymbol.createSwatch(withBackgroundColor: nil, screen: .main) { [weak self] (image, error) in
+                    if let error = error {
+                        print("Error creating swatch: \(error)")
+                    } else {
+                        self?.attachmentBBI.image = image?.withRenderingMode(.alwaysOriginal)
+                    }
+                }
+                self.connectivitySymbol.createSwatch(withBackgroundColor: nil, screen: .main) { [weak self] (image, error) in
+                    if let error = error {
+                        print("Error creating swatch: \(error)")
+                    } else {
+                        self?.connectivityBBI.image = image?.withRenderingMode(.alwaysOriginal)
+                    }
+                }
+                self.addAssociationGraphics()
+                self.mapViewDidChange()
+            }
+        }
+    }
+    
+    func addAssociationGraphics() {
+        // Check if the current viewpoint is outside of the max scale.
+        if let targetScale = mapView.currentViewpoint(with: .centerAndScale)?.targetScale {
+            if targetScale >= maxScale {
+                return
+            }
+        }
+        
+        // Check if the current viewpoint has an extent.
+        if let extent = mapView.currentViewpoint(with: .boundingGeometry)?.targetGeometry.extent {
+            //Get all of the associations in extent of the viewpoint.
+            utilityNetwork.associations(withExtent: extent) { [weak self] (associations, error) in
+                if let error = error {
+                    print("Error loading associations: \(error)")
+                } else {
+                    guard let self = self else { return }
+                    associations?.forEach { association in
```

Comment:
Please unwrap `associations` as part of the `if` statement:

```swift
} else if let associations = associations {
```

Also, please use `compactMap(_:)` (returning `nil` if a graphic isn't needed) instead of `forEach(_:)`:

```swift
let graphics = associations.compactMap { ... }
self.associationsOverlay.graphics.addObjects(from: graphics)
```